lib/web/apiserver.go

```diff
@@ -3459,6 +3478,39 @@ func (h *Handler) podConnect(
 	return nil, nil
 }
 
+func readPodExecRequestFromWS(ws *websocket.Conn) (*PodExecRequest, error) {
+	err := ws.SetReadDeadline(time.Now().Add(defaults.KubeExecDataWaitTimeout))
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to set read deadline for websocket connection")
+	}
+
+	messageType, bytes, err := ws.ReadMessage()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = ws.SetReadDeadline(time.Time{})
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to set read deadline for websocket connection")
+	}
```

Comment:
We don't enforce any read deadline for kube exec code. Also, we have ping loop going on, so if anything happens and pings stop it will close the connection?